feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/usecases/steps/BuildStepsUseCase.kt

```diff
@@ -222,14 +222,26 @@ internal class BuildStepsUseCase @Inject constructor(
         )
     )
 
-    private fun buildFetchGuidStep(projectId: String, subjectId: String) = listOf(
-        Step(
-            id = StepId.FETCH_GUID,
-            navigationActionId = R.id.action_orchestratorFragment_to_fetchSubject,
-            destinationId = FetchSubjectContract.DESTINATION,
-            payload = FetchSubjectContract.getArgs(projectId, subjectId),
-        )
-    )
+    private fun buildFetchGuidStepIfNeeded(action: ActionRequest.VerifyActionRequest) =
+        with(action) {
+            val isBiometricDataSourceSimprints =
+                BiometricDataSource.fromString(
+                    value = biometricDataSource,
+                    callerPackageName = callerPackageName,
+                ) == BiometricDataSource.Simprints
+            if (isBiometricDataSourceSimprints) {
```

Comment:
While there's nothing wrong with this approach - can we use the same structure that Alex used for identification so code is more consistent?